cdap-ui/cdap-ui-upgrade/index.js

```diff
@@ -0,0 +1,163 @@
+var yauzl = require("yauzl");
+var path = require("path");
+var mkdirp = require("mkdirp");
+
+var fs = require('fs');
+var archiver = require('archiver');
+var minimist = require('minimist');
+var Listr = require('listr');
+
+var runtimearguments = minimist(process.argv.slice(2));
+var newuizippath = runtimearguments['new-ui-zip-path'];
+var cdaphome = process.env.CDAP_HOME;
+
+var folderstozip = [
+  '/dist/',
+  '/cdap_dist/',
+  '/node_modules/',
+  '/common_dist/',
+  '/dll/',
+  '/login_dist/',
+  '/old_dist/',
+  '/server/',
+  '/templates/',
+  '/wrangler_dist/'
+];
+var filestozip = [
+  '/package.json',
+  '/server.js',
+  '/VERSION'
+];
+
+var skipUpgrade = false;
+
+function checkPreRequisite() {
+  if (!newuizippath) {
+    throw 'Missing new UI zipfile path. Please provide --new-ui-zip-path argument with the location to find new UI zip file while running the script';
```

Comment:
Missing UI Pack archive path. Please provide fully qualified path to the UI Pack archive.